src/test/java/fi/vrk/xroad/monitor/monitordata/MonitorDataHandlerTest.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * The MIT License
+ * Copyright (c) 2017, Population Register Centre (VRK)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package fi.vrk.xroad.monitor.monitordata;
+
+import fi.vrk.xroad.monitor.parser.SecurityServerInfo;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+/**
+ * Tests for {@link MonitorDataHandler}
+ */
+@Slf4j
+@SpringBootTest(classes = {MonitorDataRequestBuilder.class, MonitorDataHandler.class, MonitorDataResponseParser.class})
+@RunWith(SpringRunner.class)
+public class MonitorDataHandlerTest {
+
+    @Autowired
+    private MonitorDataHandler handler;
+
+    @Autowired
+    private MonitorDataRequestBuilder request;
+
+    @Autowired
+    private MonitorDataResponseParser response;
+
+    // Test requires this to be valid and accessible server
+    private final SecurityServerInfo exampleInfo = new SecurityServerInfo(
+            "gdev-ss1.i.palveluvayla.com",
+            "http://gdev-ss1.i.palveluvayla.com",
+            "GOV",
+            "1710128-9");
+
+    @Test
+    public void makeRequest() {
+        String xmlRequest = request.getRequestXML(exampleInfo);
+        String root = handler.makeRequest(xmlRequest);
+        log.info("result: {}", root);
+        String metric = response.getMetricInformation(root);
+        log.info("body: {}", metric);
+    }
```

Comment:
There are no assertions in this test. It looks like integration test that requires access to specific server. Maybe it would be beneficial to separate this kind of tests from unit tests and use some specific switch to run it.